### PR TITLE
ACP2E-2872: [Documentation] Update documentation regarding Bundle pro…

### PR DESCRIPTION
**Pull request by arnobsh**
## Purpose of this pull request

This pull request (PR)  the documentation update for [ACP2E-2797](https://jira.corp.adobe.com/browse/ACP2E-2797). The Jira ticket link for documentation update is https://jira.corp.adobe.com/browse/ACP2E-2872

## Affected pages

Added a new section under https://developer.adobe.com/commerce/webapi/rest/modules/ called Bundle

### CODE REVIEW

**Comment by arnobsh** [comment]
@magento run all tests

File: src/pages/rest/modules/bundle/index.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+description: Create and maintain bundle product type in the Magento application.
+keywords:
+- REST
+---
+
+# Bundle module
+
+The `Bundle` module provides functionality for creating and maintaining bundle product types.
+
+## Retrieve store specific bundle product details
```
Comment (jhadobe) [inline src/pages/rest/modules/bundle/index.md:13]:
```suggestion
## Retrieve store-specific bundle product details
```

File: src/pages/rest/modules/bundle/index.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+We can retrieve store specific bundle product details for multi-store configuration
+
+The following example uses the `GET V1/products/{sku}` call to get bundle product details. For store specific
+bundle product details add `store_code` at API request. Like this `GET {store_code}/V1/products/{sku}`
```
Comment (jhadobe) [inline src/pages/rest/modules/bundle/index.md:18]:
```suggestion
The following example retrieves store-specific bundle product details for a multi-store configuration by using the `GET V1/products/{sku}` call. For store specific
bundle product details add the `store_code` to the API request. For example, `GET {store_code}/V1/products/{sku}`
```

File: src/pages/rest/modules/bundle/index.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+        {
+            "attribute_code": "category_ids",
+            "value": [
+                "32"
+            ]
+        }
+    ]
+}
+```
+We can get `option_id` from bundle product options from  the response and use the `option_id` to update bundle product option.
```
Comment (jhadobe) [inline src/pages/rest/modules/bundle/index.md:208]:
```suggestion

Retrieve the `option_id` from the bundle product options section of the response and use it to update the bundle product option.
```

Can you clarify what is being updated here and why.

File: src/pages/rest/modules/bundle/index.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+
+## Update store specific bundle product options
+
+The following example uses the `PUT /V1/bundle-products/options/{optionId}` call to update bundle product option. For store specific
+bundle product option update use `store_code` at API request. Like this `PUT {store_code}/V1/bundle-products/options/{optionId}`
+
```
Comment (jhadobe) [inline src/pages/rest/modules/bundle/index.md:214]:
```suggestion
## Update store-specific bundle product options

The following example uses the `PUT /V1/bundle-products/options/{optionId}` call to update the bundle product option. For a store-specific request, add the `store_code`. For example, `PUT {store_code}/V1/bundle-products/options/{optionId}`

```

File: src/pages/rest/modules/bundle/index.md [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+**Sample Usage:**
+
+`PUT <host>/rest/<store_code>/V1/bundle-products/options/{optionId}`
+
+## Example
+
+The following example update bundle product option:
```
Comment (jhadobe) [inline src/pages/rest/modules/bundle/index.md:221]:
```suggestion
The following example updates the bundle product option:
```

File: src/pages/rest/modules/bundle/index.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+
+```json
+14
+```
+
+## Update store specific bundle product option links
+
+The following example uses the `PUT /V1/bundle-products/{sku}/links/{optionId}` call to update bundle product option link. For store specific
+bundle product option link update use `store_code` at API request. Like this `PUT {store_code}/V1/bundle-products/{sku}/links/{optionId}`
+
```
Comment (jhadobe) [inline src/pages/rest/modules/bundle/index.md:277]:
```suggestion
## Update store-specific bundle product option links

The following example uses the `PUT /V1/bundle-products/{sku}/links/{optionId}` call to update the bundle product option link. For a store-specific call, add `store_code` to the API request. For example, `PUT {store_code}/V1/bundle-products/{sku}/links/{optionId}`

```

File: src/pages/rest/modules/bundle/index.md [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+**Sample Usage:**
+
+`PUT <host>/rest/<store_code>/V1/bundle-products/{sku}/links/{optionId}`
+
+## Example
+
+The following example update bundle product option link:
```
Comment (jhadobe) [inline src/pages/rest/modules/bundle/index.md:284]:
```suggestion
The following example updates the bundle product option link:
```

File: src/pages/rest/modules/bundle/index.md [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+    "extension_attributes": { }
+  }
+}
+```
+
+#### Response
+
+```json
+true
+```
```
Comment (jhadobe) [inline src/pages/rest/modules/bundle/index.md:313]:
```suggestion
```

```

**Review by jhadobe — CHANGES_REQUESTED** [review]

**Comment by arnobsh** [comment]
@magento run all tests

**Comment by jhadobe** [comment]
looks good to me. @keharper any thoughts?

**Comment by keharper** [comment]
@arnobsh This new topic should be step 4 of the bundle product tutorial under `rest/tutorials/bundle-product`. 

The intro paragraph should state that the actions shown below modify the bundle product in a single store instead of all stores, and that it is optional.

The first ordered list item should be the GET call. Instruct the reader to note the important fields that will be used in the subsequent items.

Then, with second and third ordered list items, show how to change the product details and options.

Remove the changes in `src/pages/rest/modules/index.md`.

Add navigation to the new topic in `src/data/navigation/sections/rest.js`, below line 406

**Comment by arnobsh** [comment]
> @arnobsh This new topic should be step 4 of the bundle product tutorial under `rest/tutorials/bundle-product`.
> 
> The intro paragraph should state that the actions shown below modify the bundle product in a single store instead of all stores, and that it is optional.
> 
> The first ordered list item should be the GET call. Instruct the reader to note the important fields that will be used in the subsequent items.
> 
> Then, with second and third ordered list items, show how to change the product details and options.
> 
> Remove the changes in `src/pages/rest/modules/index.md`.
> 
> Add navigation to the new topic in `src/data/navigation/sections/rest.js`, below line 406

Ok, Thank you. Working on it

**Comment by arnobsh** [comment]
@magento run all tests

**Comment by arnobsh** [comment]
@keharper can you please have a review on updated changes? Thanks

**Comment by keharper** [comment]
@arnobsh I meant one new topic with three steps

1.  Get bundle product details
2. Update bundle product options
3. Update bundle product links

I looked at the example for updating options, and it seems the call is updating the quantities of the two products from 1 to 10 and 12, respectively. Would it make more sense to change the prices? I don't understand why the quantities would change that drastically.

Also, you mention  retrieving the `selection_id` from the bundle product option `product_links` section, but it doesn't appear in the GET response, as far as I can tell.

**Comment by arnobsh** [comment]
@magento run all tests

**Comment by arnobsh** [comment]
> @arnobsh I meant one new topic with three steps
> 
> 1. Get bundle product details
> 2. Update bundle product options
> 3. Update bundle product links
> 
> I looked at the example for updating options, and it seems the call is updating the quantities of the two products from 1 to 10 and 12, respectively. Would it make more sense to change the prices? I don't understand why the quantities would change that drastically.
> 
> Also, you mention retrieving the `selection_id` from the bundle product option `product_links` section, but it doesn't appear in the GET response, as far as I can tell.

Updated the documents according to the review. Can you please review again and check if anything needs to be updated/changed? Thanks

**Review by keharper — APPROVED** [review]

**Pull request merged**